src/dtos/unit-control.dto.ts

```diff
@@ -16,31 +23,49 @@ export class UnitControlBaseDTO {
   //   example: propertyMetadata.parameterCode.example,
   //   name: propertyMetadata.parameterCode.fieldLabels.value,
   // })
-  @IsNotEmpty()
-  @IsInDbValues(
-    `SELECT distinct controlequipparamcode as "value" FROM camdecmpsmd.vw_unitcontrol_master_data_relationships`,
-    {
-      message: (args: ValidationArguments) => {
-        return `${args.property} [UNITCONTROL-FATAL-B] The value for ${args.value} in the Unit Control record ${args.property} is invalid`;
-      },
-    },
-  )
+  @IsNotEmpty({
+    message: (args: ValidationArguments) => {
+      return CheckCatalogService.formatResultMessage('CONTROL-1-A', {
+        fieldname: args.property,
+        key: KEY,
+      });
+    }
+  })
+  @IsValidCode(ParameterCode, {
```

Comment:
```suggestion
  @IsValidCode(ParameterCode, {
```

Previously it was doing look up for parameter on camdecmpsmd.vw_unitcontrol_master_data_relationships.
Now you are doing against parameterCode table. Also in the checks it says `Locate ParameterCode in the Control Code lookup table.`

Reply:
Follow the DB Script.